lib/web_ui/dev/integration_tests_manager.dart

```diff
@@ -391,11 +500,95 @@ class SafariIntegrationArguments extends IntegrationArguments {
+
+  /// If target name is provided integration tests can run that one test
+  /// instead of running all the tests.
+  String testTarget;
+
+  /// The build mode to run the integration tests.
+  ///
+  /// If not specified, these tests will run using 'debug, profile, release'
+  /// modes on Chrome and will run using 'profile, release' on other browsers.
+  ///
+  /// In order to skip a test for one of the modes, add the test to the
+  /// `blockedTestsListsMapForModes` list for the relevant compile mode.
+  String buildMode;
+
+  /// Whether to use html or canvaskit backend.
+  ///
+  /// If not set html rendering backend will be used.
+  String renderingBackend;
+
+  void populateOptions(ArgParser argParser) {
+    argParser
+      ..addOption(
+        'target',
+        defaultsTo: '',
+        help: 'By default integration tests are run for all the tests under'
+            'flutter/e2etests/web directory. If a test name is specified, that '
+            'only that test will run. The test name will be the name of the '
+            'integration test (e2e test) file. For example: '
+            'text_editing_integration.dart or '
+            'profile_diagnostics_integration.dart',
+      )
+      ..addOption('build-mode',
+          defaultsTo: '',
+          help: 'Flutter supports three modes when building your app. This '
+              'option sets the build mode for the integration tests. '
+              'By default an integration test will sequentially run on '
+              'multiple modes. All three modes (debug, release, profile) are '
+              'used for Chrome. Only profile, release modes will be used for '
+              'other browsers. In other words, if a build mode is selected '
+              'tests will only be run using that mode. '
+              'See https://flutter.dev/docs/testing/build-modes for more '
+              'details on the build modes.')
+      ..addOption('rendering-backend',
+          defaultsTo: '',
+          help: 'By default both `html` and `canvaskit` rendering backends are '
+              ' tested when running integration tests. If this option is set '
+              ' only one of these backends will be used. `canvaskit` and `html`'
+              ' are the only available options. ');
+  }
+
+  /// Populate browser with results of the arguments passed.
+  void parseOptions(ArgResults argResults) {
+    testTarget = argResults['target'] as String;
+    buildMode = argResults['build-mode'] as String;
+    if (!buildMode.isEmpty &&
+        buildMode.toLowerCase() != 'debug' &&
+        buildMode.toLowerCase() != 'profile' &&
+        buildMode.toLowerCase() != 'release') {
+      throw ArgumentError('Unexpected build mode: $buildMode');
+    }
+    renderingBackend = argResults['rendering-backend'] as String;
+    if (!renderingBackend.isEmpty &&
+        renderingBackend.toLowerCase() != 'html' &&
+        renderingBackend.toLowerCase() != 'canvaskit') {
```

Comment:
What happens if I pass `cAnVaSkIt`? It seems this code allows it, but then it will actually fail when we attempt to run `flutter drive` later on. Let's not use `toLowerCase`, for consistency with the flutter CLI and to enforce a single style of specifying the renderer. Different styles will complicate code search as many text search tools are case-sensitive.

Reply:
I see. I wish other tools also allow small caps related typos but if it's the case now, I can remove the lower case conversion.

Reply:
I'm also removing `toLowerCase` for build modes. I made a quick test build modes also does not recognize case errors:

```
> flutter run -d web-server --Release
Could not find an option named "Release".


Run 'flutter -h' (or 'flutter <command> -h') for available flutter commands and options.
```

Reply:
done